src/components/UploadButton/UploadButton.tsx

```diff
@@ -18,6 +24,10 @@ export function UploadButton(props: { handleClick: (args?: any) => void }) {
       return;
     }
 
+    const previewSource = URL.createObjectURL(fileObj);
+
+    props.handleFileChange(fileObj, previewSource, true);
```

Comment:
Weak opinion but having two `handleFileChange` functions here might be a bit confusing